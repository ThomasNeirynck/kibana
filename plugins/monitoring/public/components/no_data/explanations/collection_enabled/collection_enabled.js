import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  EuiCode,
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiLoadingSpinner
} from '@elastic/eui';

const WaitingIndicator = ({ isCollectionEnabledUpdating }) => {
  return isCollectionEnabledUpdating ? <EuiLoadingSpinner size="m" /> : null;
};

export class ExplainCollectionEnabled extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { enabler } = this.props;
    enabler.enableCollectionEnabled();
  }

  render() {
    const {
      context,
      property,
      data,
      isCollectionEnabledUpdated,
      isCollectionEnabledUpdating
    } = this.props;

    const renderButton = () => (
      <EuiFlexGroup alignItems="center" gutterSize="s">
        <EuiFlexItem grow={false}>
          <EuiButton
            fill={true}
            onClick={this.handleClick}
            type="button"
            data-test-subj="enableCollectionEnabled"
          >
            Enable Monitoring Collection
          </EuiButton>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <WaitingIndicator
            isCollectionEnabledUpdating={isCollectionEnabledUpdating}
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
    const renderSuccess = () => (
      <Fragment>
        <p>
          Success! An acknowledgement came back from Elasticsearch that the
          setting was set in the cluster&apos;s persistent settings.
        </p>
        <p>
          You&apos;ll need to wait just few moments for monitoring data to start
          to appear in the cluster. As soon as as that happens, this page will
          automatically redirect to your cluster information.
        </p>
      </Fragment>
    );

    // prettier-ignore
    return (
      <Fragment>
        <p>
          We checked the {context} settings and found that <EuiCode>{property}</EuiCode>
          is set to <EuiCode>{data}</EuiCode>, which disables Monitoring.
        </p>
        <p>
          Monitoring is now disabled by default since it comes installed
          automatically. Not to worry! You can enable it right here.
        </p>

        {isCollectionEnabledUpdated ? renderSuccess() : renderButton()}
      </Fragment>
    );
  }
}

ExplainCollectionEnabled.propTypes = {
  enabler: PropTypes.object.isRequired,
  context: PropTypes.string.isRequired,
  property: PropTypes.string.isRequired,
  data: PropTypes.string.isRequired,
  isCollectionEnabledUpdated: PropTypes.bool,
  isCollectionEnabledUpdating: PropTypes.bool
};
