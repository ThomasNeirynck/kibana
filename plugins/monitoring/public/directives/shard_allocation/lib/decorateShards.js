/**
 * ELASTICSEARCH CONFIDENTIAL
 * _____________________________
 *
 *  [2014] Elasticsearch Incorporated All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Elasticsearch Incorporated and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Elasticsearch Incorporated
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from Elasticsearch Incorporated.
 */
import { capitalize, find, get, includes } from 'lodash';

export default function decorateShards(shards, nodes) {
  function getTooltipMessage(shard) {
    let relocatingNode;
    relocatingNode = find(nodes, (n) => includes(n.node_ids, shard.relocating_node));
    relocatingNode = get(relocatingNode, 'name');

    if (relocatingNode) {
      if (shard.state === 'INITIALIZING') {
        return `Relocating from ${relocatingNode}`;
      }
      if (shard.state === 'RELOCATING') {
        return `Relocating to ${relocatingNode}`;
      }
    }
    return capitalize(shard.state.toLowerCase());
  }

  return shards.map((shard) => {
    const node = nodes[shard.resolver];
    shard.nodeName = (node && node.name) || null;
    shard.type = 'shard';
    shard.tooltip_message = getTooltipMessage(shard);
    return shard;
  });
};
