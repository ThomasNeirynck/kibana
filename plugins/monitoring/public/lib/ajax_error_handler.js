import uiChrome from 'ui/chrome';

export default function ajaxErrorHandlersProvider(Notifier, $window, Promise) {
  return (err) => {
    if (err.status === 403) {
      /* redirect to error message view */
      $window.location.href = uiChrome.addBasePath('/app/monitoring#/access-denied');
    } else {
      const genericNotifier = new Notifier({ location: 'Monitoring' });
      genericNotifier.fatal(err);
    }

    return Promise.reject(err);
  };
};
