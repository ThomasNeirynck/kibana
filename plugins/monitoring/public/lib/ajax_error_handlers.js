module.exports = function ajaxErrorHandlersProvider(Notifier) {
  return {
    /* full-screen error message */
    fatalError(err) {
      if (err.status === 403) {
        const shieldNotifier = new Notifier({ location: 'Security Plugin' });
        return shieldNotifier.fatal('Sorry, you are not authorized to access Monitoring');
      }
      const genericNotifier = new Notifier({ location: 'Monitoring' });
      return genericNotifier.fatal(err);
    },
    /* dismissable banner message */
    nonFatal(err) {
      const notifier = new Notifier({ location: 'Monitoring' });
      return notifier.error(err);
    }
  };
};
