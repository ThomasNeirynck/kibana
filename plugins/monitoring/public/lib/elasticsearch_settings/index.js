export { Enabler } from './enabler';
export { ClusterSettingsChecker } from './checkers/cluster_settings';
export { NodeSettingsChecker } from './checkers/node_settings';
export { startChecks } from './start_checks';
