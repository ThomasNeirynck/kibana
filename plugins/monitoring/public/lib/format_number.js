import moment from 'moment';
import 'moment-duration-format';
import numeral from 'numeral';
import {
  FORMAT_DURATION_TEMPLATE_SHORT,
  FORMAT_DURATION_TEMPLATE_LONG,
  CALCULATE_DURATION_SINCE,
  CALCULATE_DURATION_UNTIL
} from 'monitoring-constants';

export function formatBytesUsage(used, max) {
  return formatNumber(used, 'bytes') + ' / ' + formatNumber(max, 'bytes');
}

export function formatPercentageUsage(used, max) {
  return formatNumber(used / max, '0.00%');
}

/*
 * Formats a timestamp string
 * @param timestamp: ISO time string
 * @param calculationFlag: control "since" or "until" logic
 * @return string
 */
export function formatTimestampToDuration(timestamp, calculationFlag) {
  let duration;
  if (calculationFlag === CALCULATE_DURATION_SINCE) {
    duration = moment.duration(moment.utc() - moment.utc(timestamp)); // since: now - timestamp
  } else if (calculationFlag === CALCULATE_DURATION_UNTIL) {
    duration = moment.duration(moment.utc(timestamp) - moment.utc()); // until: timestamp - now
  } else {
    throw new Error(
      '[formatTimestampToDuration] requires a [calculationFlag] parameter to specify format as "since" or "until" the given time.'
    );
  }

  if (moment.utc().diff(timestamp, 'months') >= 1) {
    // time diff is greater than 1 month, show months / days
    return moment.duration(duration).format(FORMAT_DURATION_TEMPLATE_LONG);
  }
  // time diff is less than 1 month, show days / hours / minutes
  return moment.duration(duration).format(FORMAT_DURATION_TEMPLATE_SHORT);
}

export function formatNumber(num, which) {
  if (typeof num === 'undefined' || isNaN(num)) { return 0; }
  if (typeof num !== 'number') { return num; }
  let format = '0,0.0';
  let postfix = '';
  switch (which) {
    case 'time_since':
      return moment(moment() - num).from(moment(), true);
    case 'time':
      return moment(num).format('H:mm:ss');
    case 'int_commas':
      format = '0,0';
      break;
    case 'byte':
      format += 'b';
      break;
    case 'ms':
      postfix = 'ms';
      break;
    default:
      if (which) { format = which; }
  }
  return numeral(num).format(format) + postfix;
};
