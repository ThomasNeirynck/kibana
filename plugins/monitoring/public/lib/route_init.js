var _ = require('lodash');
module.exports = function routeInitProvider(Notifier, Private, monitoringClusters, globalState, license, kbnUrl) {
  var phoneHome = Private(require('plugins/monitoring/lib/phone_home'));
  var ajaxErrorHandlers = Private(require('plugins/monitoring/lib/ajax_error_handlers'));
  return function () {
    var notify = new Notifier({ location: 'Monitoring' });
    return monitoringClusters()
    .then(function (clusters) {
      return phoneHome.sendIfDue(clusters).then(() => {
        return clusters;
      });
    })
    // Set the clusters collection and currentn cluster in globalState
    .then(function (clusters) {
      const cluster = (() => {
        const existingCurrent = _.find(clusters, { cluster_uuid: globalState.cluster_uuid });
        if (existingCurrent) return existingCurrent;

        const firstCluster = _.first(clusters);
        if (firstCluster && firstCluster.cluster_uuid) return firstCluster;

        return null;
      }());

      if (cluster) {
        globalState.cluster_uuid = cluster.cluster_uuid;
        globalState.save();
      } else {
        notify.error('We can\'t seem to find any clusters in your Monitoring data. Please check your Monitoring agents');
        return kbnUrl.redirect('/no-data');
      }

      const clusterLicense = cluster.license;
      const isExpired = (new Date()).getTime() > clusterLicense.expiry_date_in_millis;

      if (isExpired && !_.contains(window.location.hash, 'license')) {
        // redirect to license, but avoid infinite loop
        // eventually this should be a redirect to Kibana Management page
        kbnUrl.redirect('/license');
      }
      license.setLicenseType(clusterLicense.type);
      return clusters;
    })
    .catch(ajaxErrorHandlers.fatalError);
  };
};
