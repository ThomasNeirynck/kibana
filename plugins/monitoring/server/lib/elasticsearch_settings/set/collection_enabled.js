export function setCollectionEnabled(req) {
  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('admin');
  const params = {
    body: {
      transient: { 'xpack.monitoring.collection.enabled': null }, // clears the disabling method used in testing environment
      persistent: { 'xpack.monitoring.collection.enabled': true }
    }
  };

  // https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-cluster-putsettings
  return callWithRequest(req, 'cluster.putSettings', params);
}
