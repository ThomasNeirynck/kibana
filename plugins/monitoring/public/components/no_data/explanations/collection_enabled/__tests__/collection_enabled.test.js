import React from 'react';
import sinon from 'sinon';
import { mount } from 'enzyme';
import { ExplainCollectionEnabled } from '../collection_enabled';
import { findTestSubject } from '@elastic/eui/lib/test';

const enabler = {};
let component;

describe('ExplainCollectionEnabled', () => {
  beforeEach(() => {
    enabler.enableCollectionEnabled = sinon.spy();
    component = (
      <ExplainCollectionEnabled
        context="cluster"
        property="xpack.monitoring.collection.enabled"
        data="-1"
        enabler={enabler}
      />
    );
  });

  test('should explain about xpack.monitoring.collection.enabled setting', () => {
    const rendered = mount(component);
    expect(rendered).toMatchSnapshot();
  });

  test('should have a button that triggers ajax action', () => {
    const rendered = mount(component);
    const actionButton = findTestSubject(rendered, 'enableCollectionEnabled');
    actionButton.simulate('click');
    expect(enabler.enableCollectionEnabled.calledOnce).toBe(true);
  });
});
