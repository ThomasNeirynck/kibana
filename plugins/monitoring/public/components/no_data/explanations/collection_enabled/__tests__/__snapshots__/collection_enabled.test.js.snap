// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplainCollectionEnabled should explain about xpack.monitoring.collection.enabled setting 1`] = `
<ExplainCollectionEnabled
  context="cluster"
  data="-1"
  enabler={
    Object {
      "enableCollectionEnabled": [Function],
    }
  }
  property="xpack.monitoring.collection.enabled"
>
  <p>
    We checked the 
    cluster
     settings and found that 
    <EuiCode>
      <EuiCodeBlockImpl
        fontSize="s"
        inline={true}
        paddingSize="l"
        transparentBackground={false}
      >
        <span
          className="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--inline"
          style={Object {}}
        >
          <code
            className="euiCodeBlock__code"
          >
            xpack.monitoring.collection.enabled
          </code>
        </span>
      </EuiCodeBlockImpl>
    </EuiCode>
    is set to 
    <EuiCode>
      <EuiCodeBlockImpl
        fontSize="s"
        inline={true}
        paddingSize="l"
        transparentBackground={false}
      >
        <span
          className="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingLarge euiCodeBlock--inline"
          style={Object {}}
        >
          <code
            className="euiCodeBlock__code"
          >
            -1
          </code>
        </span>
      </EuiCodeBlockImpl>
    </EuiCode>
    , which disables Monitoring.
  </p>
  <p>
    Monitoring is now disabled by default since it comes installed automatically. Not to worry! You can enable it right here.
  </p>
  <EuiFlexGroup
    alignItems="center"
    component="div"
    gutterSize="s"
    justifyContent="flexStart"
    responsive={true}
    wrap={false}
  >
    <div
      className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--responsive"
    >
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <EuiButton
            color="primary"
            data-test-subj="enableCollectionEnabled"
            fill={true}
            iconSide="left"
            onClick={[Function]}
            type="button"
          >
            <button
              className="euiButton euiButton--primary euiButton--fill"
              data-test-subj="enableCollectionEnabled"
              onClick={[Function]}
              type="button"
            >
              <span
                className="euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  Enable Monitoring Collection
                </span>
              </span>
            </button>
          </EuiButton>
        </div>
      </EuiFlexItem>
      <EuiFlexItem
        component="div"
        grow={false}
      >
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <WaitingIndicator />
        </div>
      </EuiFlexItem>
    </div>
  </EuiFlexGroup>
</ExplainCollectionEnabled>
`;
